pkg/deployment/all_in_one_test.go

```diff
@@ -91,6 +91,42 @@ func TestAllInOneLabels(t *testing.T) {
 	assert.Equal(t, "false", dep.Spec.Selector.MatchLabels["another"])
 }
 
+func TestAllInOneOverwrittenDefaultLabels(t *testing.T) {
+	jaeger := v1.NewJaeger(types.NamespacedName{Name: "TestAllInOneOverwrittenDefaultLabels"})
+	jaeger.Spec.Labels = map[string]string{
+		"name":                   "operator",
+		"hello":                  "jaeger",
+		"app.kubernetes.io/name": "my-jaeger", // Override default labels
+	}
+	jaeger.Spec.AllInOne.Labels = map[string]string{
+		"hello":   "world", // Override top level annotation
+		"another": "false",
+	}
+
+	allinone := NewAllInOne(jaeger)
+	dep := allinone.Get()
+
+	assert.Equal(t, "operator", dep.Spec.Template.Labels["name"])
+	assert.Equal(t, "world", dep.Spec.Template.Labels["hello"])
+	assert.Equal(t, "false", dep.Spec.Template.Labels["another"])
+	assert.Equal(t, "my-jaeger", dep.Spec.Template.Labels["app.kubernetes.io/name"])
```

Comment:
Should we really allow this? Don't we rely on this for the reconciliation?

Reply:
Sorry I have missed your comment in the original issue.
Maybe we could reorder the spec during `util.Merge` so that we could ensure the reserved labels not be overwritten?

Reply:
I'm really not sure what's the desired behavior. If we are not using this field internally, we should allow people to override it. The only labels are using are these:

https://github.com/jaegertracing/jaeger-operator/blob/b24e2a716cd4f4efbba617fb29025f6edcfab24d/pkg/controller/jaeger/account.go#L21-L27

@rubenvp8510, what do you think?

Reply:
I was thinking on this carefully, I think the correct behaviour should be not allowed to override the labels we relies on for the reconciliation process. All other labels can be overrides in a consistent way.

Reply:
Would you be able to update this PR then, @rudeigerc?

Reply:
@jpkrohling Sure. So only those two labels mentioned above should not be overwritten?

Reply:
Yes. Actually, does it require a change at all? This is already the behavior of this PR, I suppose.

Reply:
Or removing the reserved entries from the CR in the normalize operation?

https://github.com/jaegertracing/jaeger-operator/blob/050e95a671d5cfceda848055d61af0968f5f68a3/pkg/strategy/controller.go#L54-L55

Reply:
Your solution sounds better. 😂
I will take a look at it. Thanks for your guidance.

Reply:
@rubenvp8510 @jpkrohling Hi, I've made an update. Please have a look at it.